typespec-extension/src/external-schemas.ts

```diff
@@ -108,53 +115,164 @@ export function createPollOperationDetailsSchema(schemas: Schemas, stringSchema:
 
 const fileDetailsMap: Map<string, ObjectSchema> = new Map();
 
+function getFileSchemaName(baseName: string) {
+  // make sure suffix "FileDetails"
+  if (baseName.toLocaleLowerCase().endsWith("filedetails")) {
+    return pascalCase(baseName);
+  } else if (baseName.toLocaleLowerCase().endsWith("file")) {
+    return pascalCase(baseName) + "Details";
+  } else {
+    return pascalCase(baseName) + "FileDetails";
+  }
+}
+
+function createFileDetailsSchema(schemaName: string, propertyName: string, namespace: string, schemas: Schemas) {
+  const fileDetailsSchema = new ObjectSchema(schemaName, 'The file details for the "' + propertyName + '" field.', {
+    language: {
+      default: {
+        namespace: namespace,
+      },
+      java: {
+        namespace: getJavaNamespace(namespace),
+      },
+    },
+    serializationFormats: [KnownMediaType.Multipart],
+  });
+  fileDetailsSchema.serializationFormats;
+  schemas.add(fileDetailsSchema);
+  fileDetailsMap.set(schemaName, fileDetailsSchema);
```

Comment:
Is this necessary? Seems all other places will set fileDetailsMap after calling this function.

Reply:
I think it is here
https://github.com/Azure/autorest.java/blob/14572ec8399685c072ac4366f889ede9ed0310f8/typespec-extension/src/external-schemas.ts#L224

If FileDetails of same name in the map, it won't get to this `createFileDetailsSchema` again.

Reply:
Yes, and we seem to have put into the map outside create function:
e.g. 
https://github.com/Azure/autorest.java/blob/14572ec8399685c072ac4366f889ede9ed0310f8/typespec-extension/src/external-schemas.ts#L227-L228

We could remove either one. Having them both seems excessive to me.

Reply:
Ah, I see. I thought `fileDetailsMap` is local var and won't be in the func. It is not.